### PR TITLE
Add Module-Lattice-Based Algorithms (ML-*) Support

### PR DESCRIPTION
**Pull request by pi-314159**
This brings support for ML-DSA and ML-KEM

### CODE REVIEW

**Comment by pi-314159** [comment]
Except for the inability to access HQC ports on the test server (which is still using a previous version of HQC and hasn't been updated), everything else works with the test server.

**Comment by baentsch** [comment]
> Except for the inability to access HQC ports on the test server (which is still using a previous version of HQC and hasn't been updated), everything else works with the test server.

@bhess fyi.

**Review by baentsch — APPROVED** [review]

File: ssl/test/runner/runner.go [hunk 1]
```diff
@@ -8537,8 +8537,8 @@ func addExtensionTests() {
 		name:     "ClientHelloPadding",
 		config: Config{
 			Bugs: ProtocolBugs{
-				// OQS note: the clienthello size of SSL_GROUP_X25519_FRODO640AES is 10018
-				RequireClientHelloSize: 10018,
+				// OQS note: the clienthello size of SSL_GROUP_X25519_FRODO640AES is 10026
+				RequireClientHelloSize: 10026,
```
[the remark this answers is not in the document]
Comment (pi-314159) [reply ssl/test/runner/runner.go:8541]: Because of the addition of MLKEM, the clienthello size gets bigger

[the remark this answers is not in the document]
Comment (baentsch) [reply ssl/test/runner/runner.go:8541]: OK... then the comment referencing Frodo640aes simply is misleading.

[the remark this answers is not in the document]
Comment (pi-314159) [reply ssl/test/runner/runner.go:8541]: The comment is correct. Please check the explanation below.

**Comment by pi-314159** [comment]
@baentsch 

ML-DSA update has changed the size of _Extension - Supported Groups_ from "00 12" to "00 14" because by default we enabled x25519_mlkem512 (and this gives the extra 2 bytes).

ML-DSA update has changed the size of _Extension - Signature Algorithms_ from "00 36" to "00 3c" because we added three signature algoithms (6 bytes).

Together, 8 bytes are added; thus, the clienthello size has increased from 10018 to 10026.

In short, the clienthello includes more KEX/SIG options; however, when we run the test, it still picks SSL_GROUP_X25519_FRODO640AES.

@bhess The HQC in oqs-provider 0.5.2 & 0.5.3 is the old one, so the test server does not need any updates now until the new versions of liboqs and oqs-provider are released.

**Comment by pi-314159** [comment]
@baentsch 

ML-KEM update has changed the size of _Extension - Supported Groups_ from "00 12" to "00 14" because by default we enabled x25519_mlkem512 (and this gives the extra 2 bytes).

ML-DSA update has changed the size of _Extension - Signature Algorithms_ from "00 36" to "00 3c" because we added three signature algoithms (6 bytes).

Together, 8 bytes are added; thus, the clienthello size has increased from 10018 to 10026.

In short, the clienthello includes more KEX/SIG options; however, when we run the test, it still picks SSL_GROUP_X25519_FRODO640AES.

@bhess The HQC in oqs-provider 0.5.2 & 0.5.3 is the old one, so the test server does not need any updates now until the new versions of liboqs and oqs-provider are released.

**Comment by baentsch** [comment]
> In short, the clienthello includes more KEX/SIG options; however, when we run the test, it still picks SSL_GROUP_X25519_FRODO640AES.

That is understood. But wouldn't it be more accurate to state "the clienthello size comprising all active client side algorithms is 10026" (possibly listing those) instead of "the clienthello size of SSL_GROUP_X25519_FRODO640AES is 10026"?

**Comment by pi-314159** [comment]
@baentsch I want to specify SSL_GROUP_X25519_FRODO640AES because if the test uses other KEX (like BIKEL), the clienthello size will change dramatically. But I like the idea, probably we can change the comment to something like this:

Any changes to signature algorithms and key exchange algorithms will result in a change in the size of the ClientHello, so the size should be recalculated every time.

**Comment by pi-314159** [comment]
@baentsch also it might not be a good idea to list all client side algorithms because there are too many.

**Comment by baentsch** [comment]
> @baentsch also it might not be a good idea to list all client side algorithms because there are too many.

Well, that actually is the very problem :-) My proposal was not to manually add all of them here (again) but the generate a list (if possible). If this list is a manual one, then forget it. If it is encoded in code somewhere, then a pointer there would be perfect. My main suggestion was to remove the misleading comment regarding Frodo as it led me astray.

**Comment by pi-314159** [comment]
@baentsch i’ll update it later.
